scripts/abins/input/tester.py

```diff
@@ -27,59 +29,79 @@ class Tester(object):
 
+        for k in range(num_k):
             freq = correct_data["datasets"]["k_points_data"]["frequencies"][str(k)]
             correct_data["datasets"]["k_points_data"]["frequencies"][str(k)] = np.asarray(freq)
 
+            if k <= max_displacement_kpt:
+                try:
+                    disp_file = f"{seedname}_atomic_displacements_data_{k}.txt"
+                    temp = np.loadtxt(abins.test_helpers.find_file(disp_file)).view(complex).reshape(-1)
+                except FileNotFoundError:
+                    raise FileNotFoundError(disp_file)
+
+                total_size = temp.size
+                num_freq = int(total_size / (atoms * 3))
+                array[str(k)] = temp.reshape(atoms, num_freq, 3)
+
         correct_data["datasets"]["k_points_data"].update({"atomic_displacements": array})
 
         return correct_data
 
     @staticmethod
-    def _cull_imaginary_modes(frequencies, displacements):
+    def _cull_imaginary_modes(frequencies: np.ndarray, displacements: Union[np.ndarray, None]):
         from abins.constants import ACOUSTIC_PHONON_THRESHOLD
 
         finite_mode_indices = frequencies > ACOUSTIC_PHONON_THRESHOLD
-        return (frequencies[finite_mode_indices], displacements[:, finite_mode_indices])
 
-    def _check_reader_data(self, correct_data=None, data=None, filename=None, extension=None):
+        if displacements is None:
+            return (frequencies[finite_mode_indices], None)
+        else:
+            return (frequencies[finite_mode_indices], displacements[:, finite_mode_indices])
+
+    def _check_reader_data(self, correct_data=None, data=None, filename=None, extension=None, max_displacement_kpt=float("Inf")):
         # check data
         correct_k_points = correct_data["datasets"]["k_points_data"]
         items = data["datasets"]["k_points_data"]
 
         for k in correct_k_points["frequencies"]:
             correct_frequencies, correct_displacements = self._cull_imaginary_modes(
-                correct_k_points["frequencies"][k], correct_k_points["atomic_displacements"][k]
+                correct_k_points["frequencies"][k], correct_k_points["atomic_displacements"].get(k, None)
             )
-            calc_frequencies, calc_displacements = self._cull_imaginary_modes(items["frequencies"][k], items["atomic_displacements"][k])
+            calc_frequencies, calc_displacements = self._cull_imaginary_modes(
+                items["frequencies"][k], items["atomic_displacements"].get(k, None)
+            )
+
             assert_allclose(correct_frequencies, calc_frequencies)
-            assert_allclose(correct_displacements, calc_displacements)
             assert_allclose(correct_k_points["k_vectors"][k], items["k_vectors"][k])
             self.assertEqual(correct_k_points["weights"][k], items["weights"][k])
 
+            if int(k) <= max_displacement_kpt:
+                assert_allclose(correct_displacements, calc_displacements)
```

Comment:
if this fails can we get a nice error message?

Reply:
I'm not sure it would make sense to do it just for this check and not for all the other `AssertEqual` and `assert_allclose`.

But there is an `err_msg` argument to `assert_allclose` so it wouldn't need to add complicated logic. What kind of information would be useful? What is a typical amount of information for Mantid unit tests?

Reply:
if it throws an error message thats good. I was just concerned that it might be difficult to understand `1 =!  3 on line 103` or something similar. A message just alluding to what the problem actually is would be good.

Reply:
`assert_allclose` gives pretty good error messages actually, it's a lot more helpful for arrays than `assertEqual` or `assertTrue(allclose(a, b))`. But I can use the `err_msg` to add context as well.